src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -734,17 +736,7 @@ async Task<bool> NextResultSchemaOnly(bool async, bool isConsuming = false, Canc
 
                 // Found a resultset
                 if (RowDescription is not null)
-                {
-                    if (ColumnInfoCache?.Length >= ColumnCount)
-                        Array.Clear(ColumnInfoCache, 0, ColumnCount);
-                    else
-                    {
-                        if (ColumnInfoCache is { } cache)
-                            ArrayPool<ColumnInfo>.Shared.Return(cache, clearArray: true);
-                        ColumnInfoCache = ArrayPool<ColumnInfo>.Shared.Rent(ColumnCount);
-                    }
                     return true;
```

Comment:
BTW, this is probably be wrong, as `RowDescription` can be null in case we try to get schema for queries which do not return a resultset (like `UPDATE`).

Reply:
Not sure I understand, isn't that the point?

Reply:
Let's say we have a batch consisting of `SELECT`, `UPDATE` and another `SELECT`. For the first query in the batch we return `true`, for the second `false`. Since it has returned `false`, does that mean that user doesn't have to call another `NextResult` since according to documentation `there are no more result sets if NextResult returns false`?